formats/protobuf/commonMain/src/kotlinx/serialization/protobuf/internal/Streams.kt

```diff
@@ -189,42 +200,33 @@ internal class ByteArrayOutput {
     }
 
     fun encodeVarint32(value: Int) {
-        ensureCapacity(5)
         // Fast-path: unrolled loop for single byte
+        ensureCapacity(5)
         if (value and 0x7F.inv() == 0) {
             array[position++] = value.toByte()
             return
         }
-        // Fast-path: unrolled loop for two bytes
-        var current = value
-        array[position++] = (current or 0x80).toByte()
-        current = current ushr 7
-        if (value and 0x7F.inv() == 0) {
-            array[position++] = value.toByte()
-            return
-        }
-        encodeVarint32SlowPath(current)
+        val length = varIntLength(value.toLong())
+        encodeVarint(value.toLong(), length)
+    }
+
+    fun encodeVarint64(value: Long) {
+        val length = varIntLength(value)
+        ensureCapacity(length + 1)
+        encodeVarint(value, length)
     }
 
-    private fun encodeVarint32SlowPath(value: Int) {
+    private inline fun encodeVarint(value: Long, length: Int) {
```

Comment:
It is unlikely that someone now is using a 32-bit CPU but these calculations can be slow on them.

Reply:
Could you please elaborate on what you mean here?

Reply:
We generally do not care about 32-bit platforms